gerrychain/tree.py

```diff
@@ -151,11 +159,37 @@ def part_nodes(start):
                                 queue.append(c)
             return nodes
 
-        if abs(tree_pop - h.ideal_pop) <= h.ideal_pop * h.epsilon:
-            cuts.append(Cut(edge=(node, pred[node]), subset=part_nodes(node)))
-        elif abs((total_pop - tree_pop) - h.ideal_pop) <= h.ideal_pop * h.epsilon:
-            cuts.append(Cut(edge=(node, pred[node]),
-                            subset=set(h.graph.nodes) - part_nodes(node)))
+        is_balanced_A = abs(tree_pop - h.ideal_pop) <= h.ideal_pop * h.epsilon
+        is_balanced_B = abs((total_pop - tree_pop) - h.ideal_pop) <= h.ideal_pop * h.epsilon
+
+        parent = pred[node]
+        if region_weights is not None:
+            region_cols = [tup[0] for tup in region_weights]
+            # score function to prefer dividing higher ranked regions. three regions -> [4, 2, 1]
```

Comment:
It would be nice to flesh out the example here a little more rather than using shorthand (e.g. `three regions -> [4, 2, 1]`)